service/apis/meta_service.go

```diff
@@ -105,3 +106,68 @@ func (s MetaService) GetMetadataFields(name string) ([]MetadataField, error) {
 	}
 	return nil, nil
 }
+
+// CheckKeys checks if keys exist
+func CheckKeys(keys []string, data map[string]interface{}) error {
+	for k := range keys {
+		if _, ok := data[keys[k]]; !ok {
+			return fmt.Errorf("%q doesn't exist", keys[k])
+		}
+	}
+	return nil
+}
+
+//SetDefaultKey sets default values for the keys if any isn't set
+func SetDefaultKey(dkMap map[string]interface{}, data map[string]interface{}) error {
+	for key := range dkMap {
+		if _, ok := data[key]; !ok {
+			data[key] = dkMap[key]
+			log.Infof("%v doesn't exist. set to default %#v", key, dkMap[key])
+		}
+	}
+	return nil
+}
+
+// CreateMetadata save new metadata to the storage
+func (s MetaService) CreateMetadata(meta string, data map[string]interface{}) (map[string]string, error) {
+	var rkeys = []string{"name", "fields", "objtype"}
+	err := CheckKeys(rkeys, data)
+	if err != nil {
+		return nil, err
+	}
+	fMap := data["fields"].([]interface{})
```

Comment:
ok for now but better use `ok` return value to check if `data["fields"]` is really of type `[]interface{}`
https://golang.org/ref/spec#Type_assertions

Reply:
ok